server/api/utils/clients/async_nuclio.py

```diff
@@ -18,35 +18,100 @@
 import aiohttp
 
 import mlrun.common.schemas
+import mlrun.errors
+import mlrun.runtimes.nuclio.api_gateway
 import mlrun.utils
+from mlrun.common.constants import MLRUN_CREATED_LABEL
 from mlrun.utils import logger
 
 NUCLIO_API_SESSIONS_ENDPOINT = "/api/sessions/"
-NUCLIO_API_GATEWAYS_ENDPOINT = "/api/api_gateways/"
-API_GATEWAY_NAMESPACE_HEADER = "X-Nuclio-Api-Gateway-Namespace"
+NUCLIO_API_GATEWAYS_ENDPOINT = "/api/api_gateways/{api_gateway}"
+NUCLIO_API_GATEWAY_NAMESPACE_HEADER = "X-Nuclio-Api-Gateway-Namespace"
 NUCLIO_PROJECT_NAME_HEADER = "X-Nuclio-Project-Name"
+NUCLIO_PROJECT_NAME_LABEL = "nuclio.io/project-name"
+IGUAZIO_USERNAME_LABEL = "iguazio.com/username"
 
 
 class Client:
     def __init__(self, auth_info: mlrun.common.schemas.AuthInfo):
         self._session = None
+        self._username = auth_info.username
```

Comment:
why do you need it explicitly here?

Reply:
@liranbg to set `iguazio.com/username` labels below